FirebaseAppDistribution/Sources/FIRAppDistributionMachO.m

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <CommonCrypto/CommonHMAC.h>
+#include <mach-o/arch.h>
+#import <mach-o/fat.h>
+#import <mach-o/loader.h>
+#import "FIRAppDistributionMachO+Private.h"
+#import "FIRAppDistributionMachOSlice+Private.h"
+
+@implementation FIRAppDistributionMachO
+
+- (instancetype)initWithPath:(NSString*)path {
+  self = [super init];
+
+  if (self) {
```

Comment:
Great question - this has kindof become a code pattern for every Objective-C constructor to propagate any problem that the parent class had in construction. If the parent class returned nil, setting instance variables in this class could cause a crash.

Being pedantic, I prefer the `if (!self) { return nil }` version haha

Reply:
Good question! It should never fail in this case (sans bugs in the code hash parsing), but if it was pointed at a file that wasn't a binary or the file was corrupt (which is unlikely in this case because the file we're pointing at is the app that's currently running this code) it could fail. 

If it is pointed at something invalid or a parsing issue occurs, this will still produce a code hash (as opposed to crashing), which could indicate that there's an update even if there isn't. Thinking about it more, we could also have some sort of "was parsing successful" method we could use to decide if we actually wanted to perform the update check. If it wasn't successful, we would never show the update dialog (which is also a bit weird but probably better)

Reply:
Sorry, I thought you were asking a different question. What Sam said 👍

Reply:
👍 Thanks!